wgsl/index.bs

```diff
@@ -5126,30 +5126,108 @@ If the `sampler_comparison` uses bilinear filtering then the returned value is
+textureRead(t : texture_storage_ro_3d<T>,         coords : vec3<u32>)                                                     -> vec4<C>
+```
+
+For [read-only storage textures](#texture-ro) the returned channel format `C`
+depends on the texel format `T`.
+[See the texel format table](#storage-texel-formats) for the mapping of texel
+format to channel format.
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`t`<td>
+  The [sampled](#sampled-texture-type),
+  [multisampled](#multisampled-texture-type), [depth](#texture-depth) or
+  [read-only storage](#texture-ro) texture.
+  <tr><td>`coords`<td>
+  The 0-based texel coordinate.
+  <tr><td>`array_index`<td>
+  The 0-based texture array index.
+  <tr><td>`level`<td>
+  The mip level, with 0 representing the largest level.
+  <tr><td>`sample_index`<td>
+  The 0-based sample index of the multisampled texture.
+</table>
+
+**Returns:**
 
-`vec4<type> textureLoad(texture_1d      , i32       coords, i32 level_of_detail)`
-`vec4<type> textureLoad(texture_2d      , vec2<i32> coords, i32 level_of_detail)`
-`vec4<type> textureLoad(texture_1d_array, vec2<i32> coords, i32 level_of_detail)`
-`vec4<type> textureLoad(texture_3d      , vec3<i32> coords, i32 level_of_detail)`
-`vec4<type> textureLoad(texture_2d_array, vec3<i32> coords, i32 level_of_detail)`
-  %32 = OpImageFetch %v4float %texture %coords Lod %level_of_detail
+If all the parameters are within bounds, the unfiltered texel data.<br>
+If any of the parameters are out of bounds, then zero in all components.
 
-`vec4<type> textureLoad(texture_multisampled_2d, vec2<i32> coords, i32 sample_index)`
-  %32 = OpImageFetch %v4float %texture_multisampled %coords Sample %sample_index
 
-TODO(dsinclair): Add `textureWrite` method for texture_storage_wo with integral coords
+### `textureWrite` ### {#texturewrite}
 
+Writes a single texel to a texture.
 
+<!-- TODO(ben-clayton): Remove parameter alignment padding -->
 
+```rust
+textureWrite(t : texture_storage_wo_1d<T>,       coords : u32,                          vec4<C> value) -> void
+textureWrite(t : texture_storage_wo_1d_array<T>, coords : u32,       array_index : u32, vec4<C> value) -> void
+textureWrite(t : texture_storage_wo_2d<T>,       coords : vec2<u32>,                    vec4<C> value) -> void
+textureWrite(t : texture_storage_wo_2d_array<T>, coords : vec2<u32>, array_index : u32, vec4<C> value) -> void
+textureWrite(t : texture_storage_wo_3d<T>,       coords : vec3<u32>,                    vec4<C> value) -> void
+```
+
+The channel format `C` depends on the storage texel format `T`.
+[See the texel format table](#storage-texel-formats) for the mapping of texel
+format to channel format.
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`t`<td>
+  The [write-only storage texture](#texture-wo).
+  <tr><td>`coords`<td>
+  The 0-based texel coordinate.<br>
+  Coordinates outside of the texture dimensions are clamped before performing
```

Comment:
Since the coordinates are unsigned, this means clamped to the maximum valid value?  Suggest saying that.

Reply:
Removed.